core/api/serializers.py

```
@@ -52,6 +43,46 @@ def validate_name(self, value):
+    is_profile_verified = serializers.BooleanField(read_only=True)
+    
+    # Write-only fields for updating with just IDs
+    university_id = serializers.UUIDField(write_only=True, required=False)
+    department_id = serializers.UUIDField(write_only=True, required=False)
+    major_subject_id = serializers.UUIDField(write_only=True, required=False)
+
+    class Meta:
+        model = CustomUser
+        fields = ['id', 'username', 'email', 'profile_picture', 'is_email_verified', 'phone_number', 'is_profile_verified', 'date_of_birth', 'university', 'department', 'major_subject', 'university_id', 'department_id', 'major_subject_id']
+    
+    def update(self, instance, validated_data):
```

Ruff (0.8.2)

Missing return type annotation for public function `update`

(ANN201)